### PR TITLE
[SYCL] Remove more program class usages

### PR DESCRIPTION
**Pull request by alexbatashev**

### CODE REVIEW

File: SYCL/DeprecatedFeatures/histogram_256_slm_spec.cpp [hunk 1]
```diff
@@ -0,0 +1,207 @@
+//==--------------- histogram_256_slm.cpp  - DPC++ ESIMD on-device test ----==//
```
Comment (kbobrovs) [inline SYCL/DeprecatedFeatures/histogram_256_slm_spec.cpp]: - The test path should have esimd somewhere (DeprecatedFeatures/esimd?)
- As I see, SYCL/ESIMD/mandelbrot/mandelbrot_spec.cpp was fixed in this PR to use 2020 spec constants, can the same be done for other tests (e.g. this one)?

**Review by vladimirlaz — DISMISSED** [review]

File: SYCL/DeprecatedFeatures/histogram_256_slm_spec.cpp (hunk 1 above)
Comment (alexbatashev) [reply SYCL/DeprecatedFeatures/histogram_256_slm_spec.cpp]: > The test path should have esimd somewhere (DeprecatedFeatures/esimd?)

Fixed

> can the same be done for other tests (e.g. this one)?

I'm not sure. Some tests have a SYCL 2020 version (like this one), and others don't, and I don't see any pattern here.

**Review by kbobrovs — DISMISSED** [review]

**Review by kbobrovs — COMMENTED** [review]
This change would make sense only if the original SYCL2020 version does not work - is this the case? Otherwise it would better just to move it back to SYCL/ESIMD.

**Comment by alexbatashev** [comment]
> This change would make sense only if the original SYCL2020 version does not work - is this the case? Otherwise it would better just to move it back to SYCL/ESIMD.

Not sure what you mean. If you're talking about histogram test, there's a very similar test with SYCL2020 APIs. For others - we're removing program class completely, so there's no option to leave everything as is. Tests won't work.

**Review by AlexeySachkov — APPROVED** [review]

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**